compiler/semcall.nim

```diff
@@ -58,6 +58,7 @@ proc initCandidateSymbols(c: PContext, headSymbol: PNode,
       ]#
       for paramSym in searchInScopesAllCandidatesFilterBy(c, symx.name, {skConst}):
         let paramTyp = paramSym.typ
+        if paramTyp.sym == nil and paramTyp.n.kind == nkIntLit: continue 
```

Comment:
Makes no sense whatsoever. Parameters with default values don't participate in overload resolution in exactly the same way as parameters without default values.

Reply:
All right, the issue is that just below it, accessing `paramTyp.n.sym.kind ` will crash even though `paramTyp.n.sym` is not nil. It seems like a deeper issue when initialising the scope symbols. Since this PR fix another issue, will remove this one and attempt to tackle down the road